pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -565,7 +565,11 @@ func (r *ReconcileNodeMaintenance) cancelEviction(drainer *drain.Helper, nodeNam
 	if len(pods) != 0 {
 		// cancel the move
 		for _, pod := range pods {
-			return drainer.Client.PolicyV1beta1().Evictions(pod.Namespace).Evict(nil)
+			err := drainer.Client.PolicyV1beta1().Evictions(pod.Namespace).Evict(nil)
```

Comment:
Have you tested this or found documentation that `Evict()` behaves like this?

Reply:
not yet tested. I don't know how to design an experiment that could prove this, as this is a very transient state. I guess that I should look at what kubelet is doing with this field in order to find out.

Reply:
1. Create a replica set with as many copies as worker nodes, and use a label selector to require the label "evictionTest"
1. Create a PDB for the replica set's pods with maxUnavailable=1
1. Add "evictionTest" to all workers except one

Now you have a set of pods that cannot be evicted (due to the PDB)
Next:

1. Run the eviction loop on a node where the replica set is active
1. See that the pod is not evicted, record the pod's yaml
1. Run `cancelEviction()` and see what changes happen to the pod's yaml